src/kudu/consensus/raft_consensus.cc

```diff
@@ -2226,6 +2226,48 @@ Status RaftConsensus::CheckBulkConfigChangeAndGetNewConfigUnlocked(
           }
           if (IsRaftConfigVoter(server_uuid, committed_config)) {
             num_voters_modified++;
+
+            // If we are in flexi-raft mode, we want to make sure that the number of
+            // voters does not dip below quorum requirements/min-rep-factor
+            // So if we have 6 voters in LEADER region and min-replication-factor/
+            // quorum = Majority(6) = 4, then in healthy state we are expecting 6 voters.
+            // We can safely remove 2 voters and we will still have 4 voters, having enough
+            // for write-availability, but we can't remove another one as that will make #voters=3
+            // which will prevent commit with a MIN-REP-FACTOR=4
+            if (FLAGS_enable_flexi_raft) {
```

Comment:
Wondering if we should restrict this to the current leader's region i.e only disallow removal of voters in current leader's region if the removal can lead to losing majority in leader's region. If one is removing a node in a non-leader region, then dropping below the majority requirement in that particular region wont hurt write availability (current or in future since promotion to that region cannot happen).

Reply:
I had thought about this, and I agree this might be a good idea.